src/V3ExecGraph.cpp

```diff
@@ -109,52 +110,122 @@ class ThreadSchedule final {
+                // Keep original name for the original thread of hierarchical task to keep
+                // dependency tracking, add '_' for the rest to differentiate them.
+                const string name
+                    = i == 0 ? mtaskp->name() : mtaskp->name() + '_' + std::to_string(i);
+                const string label = mtaskp->name() + " (" + std::to_string(startTime(mtaskp))
+                                     + ':' + std::to_string(endTime(mtaskp)) + ')'
+                                     + "\\ncost=" + std::to_string(mtaskp->cost())
+                                     + "\\npriority=" + std::to_string(mtaskp->priority());
+                const double xPos = (threadBoxWidth + horizontalGap) * index + horizontalGap;
+                const double yPos
+                    = -threadBoxHeight
+                      * static_cast<double>(threadId(mtaskp) + i * schedule.threads.size());
+                const string fillColor = i == 0 ? "white" : "lightgreen";
+                emitBlock(name, label, threadBoxWidth, threadBoxHeight, xPos, yPos, fillColor);
             }
-        }
-        const double minWidth = 2.0;
-        const auto mtaskXPos = [&](const ExecMTask* mtaskp, const double nodeWidth) {
-            const double startPosX = (minWidth * startTime(mtaskp)) / minCost;
-            return nodeWidth / minWidth + startPosX;
         };
 
-        const auto emitMTask = [&](const ExecMTask* mtaskp) {
-            const int thread = threadId(mtaskp);
-            const double nodeWidth = minWidth * (static_cast<double>(mtaskp->cost()) / minCost);
-            const double x = mtaskXPos(mtaskp, nodeWidth);
-            const int y = -thread;
-            const string label = "label=\"" + mtaskp->name() + " (" + cvtToStr(startTime(mtaskp))
-                                 + ":" + std::to_string(endTime(mtaskp)) + ")" + "\"";
-            *logp << "  " << mtaskp->name() << " [" << label << " width=" << nodeWidth << " pos=\""
-                  << x << "," << y << "!\"]\n";
+        const auto emitFork = [&](int index) {
+            const string& name = "fork_" + std::to_string(index);
+            constexpr double width = threadBoxWidth / 8;
+            const double height = threadBoxHeight * nThreads;
+            const double xPos = index * (threadBoxWidth + horizontalGap) - horizontalGap / 2;
+            const double yPos
+                = -static_cast<double>(nThreads) / 2 * threadBoxHeight + threadBoxHeight / 2;
+            emitBlock(name, "", width, height, xPos, yPos, "black");
         };
 
-        // Emit MTasks
-        for (const V3GraphVertex& vtx : graph.vertices()) {
-            if (const ExecMTask* const mtaskp = vtx.cast<const ExecMTask>()) emitMTask(mtaskp);
-        }
+        // Create columns of tasks whose execution intervals overlaps
+        int offset = 0;
+        for (const ThreadSchedule& schedule : schedules) {
+            std::vector<std::vector<const ExecMTask*>> columns = {{}};
```

Comment:
Make a "using Columns = ..." to avoid repeating dtype below.  (Could also use an auto below, if prefer.)